File: mobile/android/FUN/app/src/main/java/com/fun/app/ui/components/overlays/ContinuePrompt.kt
/**
 * FUN App - Continue Prompt
 * Shows after Episode 1 ends in Discover mode
 */

package com.fun.app.ui.components.overlays

import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import coil.compose.AsyncImage
import com.fun.app.data.models.Episode
import com.fun.app.data.models.Series
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@Composable
fun ContinuePrompt(
    series: Series,
    nextEpisode: Episode,
    onContinue: () -> Unit,
    onSkip: () -> Unit,
    modifier: Modifier = Modifier
) {
    var countdown by remember { mutableIntStateOf(10) }
    var isVisible by remember { mutableStateOf(true) }
    val scope = rememberCoroutineScope()
    
    LaunchedEffect(Unit) {
        scope.launch {
            while (countdown > 0) {
                delay(1000)
                countdown--
            }
            onSkip()
        }
    }
    
    if (isVisible) {
        Box(
        modifier = modifier
            .fillMaxSize()
            .background(Color.Black.copy(alpha = 0.8f))
            .blur(8.dp),
        contentAlignment = Alignment.Center
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(24.dp)
        ) {
            // Next Episode Thumbnail
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp)
                    .clip(RoundedCornerShape(12.dp))
            ) {
                AsyncImage(
                    model = nextEpisode.thumbnailUrl,
                    contentDescription = nextEpisode.title,
                    modifier = Modifier.fillMaxSize(),
                    contentScale = ContentScale.Crop
                )
                
                // Gradient overlay
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .background(
                            Brush.verticalGradient(
                                colors = listOf(Color.Transparent, Color.Black.copy(alpha = 0.8f))
                            )
                        )
                )
                
                // Episode info
                Column(
                    modifier = Modifier
                        .align(Alignment.BottomStart)
                        .padding(16.dp)
                ) {
                    Text(
                        text = "Next Episode",
                        style = MaterialTheme.typography.labelSmall,
                        color = Color.White.copy(alpha = 0.8f)
                    )
                    Text(
                        text = nextEpisode.title,
                        style = MaterialTheme.typography.titleMedium,
                        color = Color.White,
                        maxLines = 1
                    )
                }
            }
            
            // Title
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Text(
                    text = "Continue watching?",
                    style = MaterialTheme.typography.headlineMedium,
                    color = Color.White,
                    fontWeight = androidx.compose.ui.text.font.FontWeight.Bold
                )
                
                Text(
                    text = series.title,
                    style = MaterialTheme.typography.titleLarge,
                    color = Color.White.copy(alpha = 0.9f)
                )
                
                Text(
                    text = "Season ${nextEpisode.seasonNumber}, Episode ${nextEpisode.episodeNumber}",
                    style = MaterialTheme.typography.bodyMedium,
                    color = Color.White.copy(alpha = 0.7f)
                )
            }
            
            // Buttons
            Column(
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Button(
                    onClick = onContinue,
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(56.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = Color.Transparent
                    ),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(Icons.Default.PlayArrow, contentDescription = null)
                        Text("Continue Watching", style = MaterialTheme.typography.titleMedium)
                    }
                }
                
                OutlinedButton(
                    onClick = onSkip,
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(56.dp),
                    colors = ButtonDefaults.outlinedButtonColors(
                        contentColor = Color.White
                    ),
                    border = androidx.compose.foundation.BorderStroke(1.dp, Color.White.copy(alpha = 0.3f)),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(Icons.Default.Close, contentDescription = null)
                        Text("Keep Discovering (${countdown}s)", style = MaterialTheme.typography.titleMedium)
                    }
                }
            }
            
            // Description
            nextEpisode.description?.let { description ->
                Text(
                    text = description,
                    style = MaterialTheme.typography.bodySmall,
                    color = Color.White.copy(alpha = 0.6f),
                    textAlign = TextAlign.Center,
                    maxLines = 2,
                    modifier = Modifier.padding(horizontal = 16.dp)
                )
            }
        }
    }
}

File: mobile/ios/FUN/FUN/Views/Components/ContinuePrompt.swift
/**
 * FUN App - Continue Prompt
 * Shows after Episode 1 ends in Discover mode
 */

import SwiftUI

struct ContinuePrompt: View {
    let series: Series
    let nextEpisode: Episode
    let onContinue: () -> Void
    let onSkip: () -> Void
    
    @State private var countdown = 10
    @State private var isVisible = true
    
    var body: some View {
        if isVisible {
            ZStack {
                // Background blur
                Color.black.opacity(0.8)
                    .ignoresSafeArea()
                    .blur(radius: 20)
                
                VStack(spacing: 24) {
                    // Next Episode Thumbnail
                    AsyncImage(url: URL(string: nextEpisode.thumbnailUrl)) { image in
                        image
                            .resizable()
                            .aspectRatio(16/9, contentMode: .fill)
                    } placeholder: {
                        Rectangle()
                            .fill(Color.gray.opacity(0.3))
                    }
                    .frame(height: 200)
                    .cornerRadius(12)
                    .overlay(
                        LinearGradient(
                            colors: [.clear, .black.opacity(0.8)],
                            startPoint: .top,
                            endPoint: .bottom
                        )
                        .cornerRadius(12)
                    )
                    .overlay(
                        VStack {
                            Spacer()
                            HStack {
                                VStack(alignment: .leading, spacing: 4) {
                                    Text("Next Episode")
                                        .font(.caption)
                                        .foregroundColor(.white.opacity(0.8))
                                    Text(nextEpisode.title)
                                        .font(.headline)
                                        .foregroundColor(.white)
                                        .lineLimit(1)
                                }
                                Spacer()
                            }
                            .padding()
                        }
                    )
                    
                    // Text
                    VStack(spacing: 8) {
                        Text("Continue watching?")
                            .font(.title2.bold())
                            .foregroundColor(.white)
                        
                        Text(series.title)
                            .font(.title3)
                            .foregroundColor(.white.opacity(0.9))
                        
                        Text("Season \(nextEpisode.seasonNumber), Episode \(nextEpisode.episodeNumber)")
                            .font(.subheadline)
                            .foregroundColor(.white.opacity(0.7))
                    }
                    
                    // Buttons
                    VStack(spacing: 12) {
                        Button(action: {
                            isVisible = false
                            DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
                                onContinue()
                            }
                        }) {
                            HStack {
                                Image(systemName: "play.fill")
                                Text("Continue Watching")
                            }
                            .font(.headline)
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .frame(height: 56)
                            .background(
                                LinearGradient(
                                    colors: [Color.purple, Color.pink],
                                    startPoint: .leading,
                                    endPoint: .trailing
                                )
                            )
                            .cornerRadius(12)
                        }
                        
                        Button(action: {
                            isVisible = false
                            DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
                                onSkip()
                            }
                        }) {
                            HStack {
                                Image(systemName: "xmark")
                                Text("Keep Discovering (\(countdown)s)")
                            }
                            .font(.headline)
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .frame(height: 56)
                            .background(Color.white.opacity(0.2))
                            .cornerRadius(12)
                            .overlay(
                                RoundedRectangle(cornerRadius: 12)
                                    .stroke(Color.white.opacity(0.3), lineWidth: 1)
                            )
                        }
                    }
                    
                    // Description
                    if let description = nextEpisode.description {
                        Text(description)
                            .font(.caption)
                            .foregroundColor(.white.opacity(0.6))
                            .multilineTextAlignment(.center)
                            .lineLimit(2)
                            .padding(.horizontal)
                    }
                }
                .padding(.horizontal, 24)
            }
            .transition(.opacity.combined(with: .scale))
            .onAppear {
                startCountdown()
            }
        }
    }
    
    private func startCountdown() {
        Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true) { timer in
            countdown -= 1
            if countdown <= 0 {
                timer.invalidate()
                onSkip()
            }
        }
    }
}
